ui/lib/use-handle-error.js

```diff
@@ -1,10 +1,13 @@
 import { useEffect } from 'react'
 import { useErrorContext } from '../components/ErrorProvider'
 
-export function useHandleError([{ data, error, loading }, method]) {
+export function useHandleError(
+  [{ data, error, loading }, method],
+  throwOnRevert = true // for some contract interactions, a reverted call is not an error
+) {
   const errorContext = useErrorContext()
   useEffect(() => {
-    errorContext.addError([error])
+    if (throwOnRevert) errorContext.addError([error])
```

Comment:
Suggestion for better readability: Use a separate line for the code inside the if statement:

```javascript
if (throwOnRevert) {
  errorContext.addError([error])
}
```